x-pack/plugins/observability_solution/observability/public/pages/alert_details/components/header_actions.tsx

```diff
@@ -109,9 +124,67 @@ export function HeaderActions({
     setSnoozeModalOpen(true);
   };
 
+  const { mutateAsync: createInvestigation } = useCreateInvestigation();
+
+  const alertStart = alert?.fields[ALERT_START];
+  const alertEnd = alert?.fields[ALERT_END];
+
+  const createOrOpenInvestigation = async () => {
+    if (!alert) return;
+
+    if (!investigations || investigations.results.length === 0) {
+      const paddedAlertTimeRange = getPaddedAlertTimeRange(alertStart!, alertEnd);
+
+      const investigationResponse = await createInvestigation({
+        investigation: {
+          id: uuidv4(),
+          title: `Investigate ${alert.fields[ALERT_RULE_CATEGORY]} breached`,
+          params: {
+            timeRange: {
+              from: new Date(paddedAlertTimeRange.from).getTime(),
+              to: new Date(paddedAlertTimeRange.to).getTime(),
+            },
+          },
+          origin: {
+            type: 'alert',
+            id: alert.fields[ALERT_UUID],
+          },
+        },
+      });
+
+      navigateToApp('investigate', { path: `/${investigationResponse.id}`, replace: false });
+    } else {
+      navigateToApp('investigate', {
+        path: `/${investigations.results[0].id}`,
+        replace: false,
+      });
+    }
+  };
+
   return (
     <>
       <EuiFlexGroup direction="row" gutterSize="s" justifyContent="flexEnd">
+        {investigate &&
```

Comment:
Is `investigate` null when the feature flag is off?

Reply:
it's `undefined`